diff --git a/dlls/winmm/waveform.c b/dlls/winmm/waveform.c
--- a/dlls/winmm/waveform.c
+++ b/dlls/winmm/waveform.c
@@ -3661,6 +3661,17 @@ UINT WINAPI mixerOpen(LPHMIXER lphMix, UINT uDeviceID, DWORD_PTR dwCallback,
     if(mr != MMSYSERR_NOERROR)
         return mr;
 
+    switch(fdwOpen & 0xF0000000){
+    case MIXER_OBJECTF_WAVEOUT:
+    case MIXER_OBJECTF_WAVEIN:
+    case MIXER_OBJECTF_HWAVEOUT:
+    case MIXER_OBJECTF_HWAVEIN:
+        mr = mixerGetID((HMIXEROBJ)(UINT_PTR)uDeviceID, &uDeviceID, fdwOpen & 0xF0000000);
+        if(mr != MMSYSERR_NOERROR)
+            return mr;
+        break;
+    }
+
     if(uDeviceID >= g_outmmdevices_count + g_inmmdevices_count)
         return MMSYSERR_BADDEVICEID;
 
